tests/components/husqvarna_automower/conftest.py

```diff
@@ -82,20 +83,22 @@ async def setup_credentials(hass: HomeAssistant) -> None:
 @pytest.fixture
 def mock_automower_client() -> Generator[AsyncMock, None, None]:
     """Mock a Husqvarna Automower client."""
-    with patch(
-        "homeassistant.components.husqvarna_automower.AutomowerSession",
-        autospec=True,
-    ) as mock_client:
-        client = mock_client.return_value
-        client.get_status.return_value = mower_list_to_dictionary_dataclass(
-            load_json_value_fixture("mower.json", DOMAIN)
-        )
 
-        async def websocket_connect() -> ClientWebSocketResponse:
-            """Mock listen."""
-            return ClientWebSocketResponse
+    async def websocket_connect() -> ClientWebSocketResponse:
+        """Mock listen."""
+        return ClientWebSocketResponse
+
+    mower_dict = mower_list_to_dictionary_dataclass(
+        load_json_value_fixture("mower.json", DOMAIN)
+    )
 
-        client.auth = AsyncMock(side_effect=websocket_connect)
-        client.commands = AsyncMock()
+    mock = AsyncMock(spec=AutomowerSession)
```

Comment:
```suggestion
    mock = AsyncMock(spec_set=AutomowerSession)
```
It's more stricter

Reply:
Probably too strict. I got this error.
```
    @pytest.fixture
    def mock_automower_client() -> Generator[AsyncMock, None, None]:
        """Mock a Husqvarna Automower client."""
    
        mower_dict = mower_list_to_dictionary_dataclass(
            load_json_value_fixture("mower.json", DOMAIN)
        )
    
        mock = AsyncMock(spec_set=AutomowerSession)
>       mock.auth = AsyncMock(side_effect=ClientWebSocketResponse)

tests/components/husqvarna_automower/conftest.py:92: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <AsyncMock spec_set='AutomowerSession' id='132223591495056'>, name = 'auth', value = <AsyncMock id='132223426177856'>

/home/vscode/.local/ha-venv/lib/python3.12/site-packages/coverage/inorout.py:519: CoverageWarning: Module homeassistant.components.husqvarna_automower was previously imported, but not measured (module-not-measured)
  self.warn(msg, slug="module-not-measured")
    def __setattr__(self, name, value):
        if name in _allowed_names:
            # property setters go through here
            return object.__setattr__(self, name, value)
        elif (self._spec_set and self._mock_methods is not None and
            name not in self._mock_methods and
            name not in self.__dict__):
>           raise AttributeError("Mock object has no attribute '%s'" % name)
E           AttributeError: Mock object has no attribute 'auth'

/usr/local/lib/python3.12/unittest/mock.py:774: AttributeError
```
But I can't figure out the problem